### PR TITLE
Pagination of library keep list

### PR DESCRIPTION
**Pull request by andrewconner**
@ahsu1230 @yipingliao

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/libraries/library.js [hunk 1]
```diff
@@ -39,36 +43,33 @@ angular.module('kifi')
...
-      return 'Showing the ' + numShown + ' latest Keeps';
+      return 'Showing ' + numShown + ' of ' + $scope.library.numKeeps + ' latest Keeps';
     };
 
     $scope.scrollDistance = '100%';
     $scope.scrollDisabled = false;
 
     $scope.getNextKeeps = function () {
-      if ($scope.loading) {
+      if ($scope.loading || !$scope.library || $scope.keeps.length == 0) {
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.js]: Use `===`.

File: kifi-backend/modules/shoebox/angular/src/libraries/library.js [hunk 2]
```diff
@@ -12,16 +12,20 @@ angular.module('kifi')
 
     var libraryP = libraryService.getLibraryByUserSlug(username, librarySlug);
 
+    $scope.loading = true;
     libraryP.then(function (library) {
+      _.forEach(library.keeps, keepService.buildKeep);
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.js:17]: I just made a PR that should make this call to `buildKeep` unnecessary.

File: kifi-backend/modules/shoebox/angular/src/libraries/library.js [hunk 3]
```diff
@@ -39,36 +43,33 @@ angular.module('kifi')
...
         return;
       }
 
       $scope.loading = true;
-      return libraryService.getLibraryByUserSlug(username, librarySlug).then(function (list) {
+      return libraryService.getKeepsInLibrary($scope.library.id, $scope.keeps.length).then(function (res) {
         $scope.loading = false;
+        // $scope.scrollDisabled = true;
+        if (res.keeps) {
+          _.forEach(res.keeps, keepService.buildKeep);
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.js:62]: Another place where `buildKeep` call may not be needed.

**Comment by yipingliao** [comment]
:+1:

File: kifi-backend/modules/shoebox/angular/src/libraries/library.js (hunk 2 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.js:17]: Thanks! I actually still want the augmentation for now (for the other fields), but will keep that in mind.

File: kifi-backend/modules/shoebox/angular/src/libraries/library.js (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.js]: Thanks!

**Pull request merged**